library/azure_rm_cosmosdbaccount.py

```diff
@@ -0,0 +1,548 @@
+        super(AzureRMCosmosDBAccount, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                      supports_check_mode=True,
+                                                      supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                self.parameters[key] = kwargs[key]
+
+        dict_camelize(self.parameters, ['kind'], True)
+        dict_camelize(self.parameters, ['consistency_policy', 'default_consistency_level'], True)
+        dict_rename(self.parameters, ['geo_rep_locations', 'name'], 'location_name')
+        dict_rename(self.parameters, ['geo_rep_locations'], 'locations')
+
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(CosmosDB,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        resource_group = self.get_resource_group(self.resource_group)
+
+        if "location" not in self.parameters:
+            self.parameters["location"] = resource_group.location
+
+        old_response = self.get_databaseaccount()
+
+        if not old_response:
+            self.log("Database Account instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Old instance didn't exist")
+            else:
+                self.to_do = Actions.Create
+        else:
+            self.log("Database Account instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                old_response['locations'] = old_response['failover_policies']
+                if not default_compare(self.parameters, old_response, '', self.results):
+                    self.to_do = Actions.Update
+
+        if (self.to_do == Actions.Create) or (self.to_do == Actions.Update):
+            self.log("Need to Create / Update the Database Account instance")
+
+            if self.check_mode:
+                self.results['changed'] = True
+                return self.results
+
+            response = self.create_update_databaseaccount()
+
+            self.results['changed'] = True
+            self.log("Creation / Update done")
+        elif self.to_do == Actions.Delete:
+            self.log("Database Account instance deleted")
+            self.results['changed'] = True
+
+            if self.check_mode:
+                return self.results
+
+            self.delete_databaseaccount()
+            # make sure instance is actually deleted, for some Azure resources, instance is hanging around
+            # for some time after deletion -- this should be really fixed in Azure.
+            while self.get_databaseaccount():
+                time.sleep(20)
```

Comment:
should there be some exit criteria of timeout?  what's the impact if deletion not completed, and the resource hang around for a while? shouldn't the poll wait  logic in ansible playbook, but not in module?

Reply:
I think it's actually not necessary for cosmosdb at all. I can just remove it.
it's  a pattern that stayed since early modules